### PR TITLE
Update deployment completeness documentation

### PR DESCRIPTION
**Pull request by kargakis**
@janetkuo PTAL

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2347)

### CODE REVIEW

File: docs/user-guide/deployments.md [hunk 1]
```diff
@@ -567,6 +567,7 @@ Kubernetes marks a Deployment as _progressing_ when one of the following tasks i
 * The Deployment is in the process of creating a new ReplicaSet.
 * The Deployment is scaling up an existing ReplicaSet.
 * The Deployment is scaling down an existing ReplicaSet.
+* New pods become available.
```
Comment (kargakis) [inline docs/user-guide/deployments.md]: Added in https://github.com/kubernetes/kubernetes/pull/36648.

File: docs/user-guide/deployments.md [hunk 2]
```diff
@@ -578,6 +579,7 @@ Kubernetes marks a Deployment as _complete_ when it has the following characteri
 equals or exceeds the number required by the Deployment strategy.
 * All of the replicas associated with the Deployment have been updated to the latest version you've specified, meaning any
 updates you've requested have been completed.
+* No old pods for the Deployment are running.
```
Comment (kargakis) [inline docs/user-guide/deployments.md]: Added in https://github.com/kubernetes/kubernetes/pull/40261

File: docs/user-guide/deployments.md (hunk 1 above)
Comment (janetkuo) [inline docs/user-guide/deployments.md]: It's not in 1.5?

Comment (kargakis) [inline docs/user-guide/deployments.md]: Not this one. Should I open a PR with the other change against the 1.5 branch and keep this as is?

Comment (janetkuo) [inline docs/user-guide/deployments.md]: Not necessary. I asked this just because current docs should only mention 1.5 features/behavior.

Comment (kargakis) [inline docs/user-guide/deployments.md]: When will a 1.6 branch will be cut? I can remove this now but don't wont to forget it. Maybe an issue?

**Comment by chenopis** [comment]
@kargakis any resolution for @janetkuo comments?

File: docs/user-guide/deployments.md (hunk 1 above)
Comment (chenopis) [inline docs/user-guide/deployments.md]: I can tag this PR so that it's not merged until the 1.6 branch has been opened.

Comment (kargakis) [inline docs/user-guide/deployments.md]: Opened https://github.com/kubernetes/kubernetes.github.io/pull/2415 for the 1.6 change.

**Comment by chenopis** [comment]
@kargakis @janetkuo So, is this ready to be merged now?

**Comment by janetkuo** [comment]
@chenopis the updated doc mentioned a change related to https://github.com/kubernetes/kubernetes/pull/40625. https://github.com/kubernetes/kubernetes/pull/40625 is already merged to 1.5 branch, but the 1.5 patch with this change isn't released yet. It may be okay to merge it now, or we can wait for after the next 1.5 patch release (1.5.3).

**Comment by chenopis** [comment]
@janetkuo Okay, I'll hold off until the 1.5.3 patch release, then merge it. Thx

**Comment by chenopis** [comment]
@janetkuo Do you know when the 1.5.3 patch release slated to go out?

**Comment by janetkuo** [comment]
> Do you know when the 1.5.3 patch release slated to go out?

I don't. 1.5 release czar @saad-ali would know

**Comment by saad-ali** [comment]
> I don't. 1.5 release czar @saad-ali would know

Soon: https://groups.google.com/forum/#!topic/kubernetes-dev/w7ifXmvxKrI

**Comment by chenopis** [comment]
Ok, cool. I'll keep tabs on it. Thx!

**Comment by chenopis** [comment]
Looks like 1.5.3 went out yesterday, so I'm merging this.

**Pull request merged**